CONTRIBUTING.md

```diff
@@ -0,0 +1,137 @@
+# Contributing guidelines
+
+## Principles and vision
+
+### [Roadmap to `v1.1.x`](./ROADMAP.md)
+
+### go version
+
+This repository supports _at least_ the 2 latest stable versions of the language.
+
+Example: when `go1.21` is out, we may require `go1.19`
+
+Any new language feature that is introduced should be guarded (and ideally, emulated for backward compatibility)
+by go build flags, e.g. `go:build !go1.21`, `go:build go1.21`.
+
+We don't upgrade the required version of the compiler automatically:
+changes to the go version requirement in `go.mod` will occur only if a new language feature is deemed really useful.
+
+Therefore, at any given time, older versions of the compiler might be supported. This not a promise, just a likely outcome.
+
+Example: if we introduce generics, `go1.18` would be a requirement.
+
+### The `pflag` compatibility promise
+
+**Programs using older versions of `pflag` should be able to compile in the future.**
+
+> In the spirit of the `go` language, we believe that the `golang` eco-system would benefit from a similar promise by this repository.
+
+The required version of the compiler may however evolve.
+
+We may use deprecation notices to advise users to move towards newer APIs.
+
+Older APIs will continue being honored, though.
+
+### The `pflag` promise of being a drop-in replacement of `flag`
+
+`pflag` is a drop-in replacement for `go`'s flag package, implementing
+POSIX/GNU-style --flags. That's it.
+
+`pflag` will follow the evolution of the standard library `flag`, and continue to maintain the promise of being
+a _drop-in replacement_ for `flag` (and then some of course, but at least that).
+
+### Versioning
+
+This package abides by `semver`.
+
+Hence:
+* `v1.0.6` : patch
+* `v1.1.0` : new feature
+
+As stated above, it is unlikely we ever issue a `v2.0.0` (breaking change).
+
+## Pull requests
+
+All contributions are much appreciated and welcome. There is no small contributions: additional documentation, examples, 
+tests, etc are all valuable additions, not just new features.
+
+### Linting
+
+This repository uses `github.com/golangci/golangci-lint/cmd/golangci-lint` as meta-linter, enforced by CI.
+The set of rules may evolve over time. `revive` and `govet` are must-have.
+
+> TODO: agree on the initial .golangci.yml setup, with a no-nonsense approach to linting. Code quality without nitpicking... :)
+> As a started, here is a sample configuration that I use on a daily basis for personal projects (no strong agenda on this):
+> [golangci-lint.yml](https://github.com/fredbi/gflag/blob/master/.golangci.yml).
```

Comment:
Linters enabled in Viper: https://github.com/spf13/viper/blob/master/.golangci.yaml

Though I'm not sure all of them is necessary.

Reply:
@sagikazarmark yeah I am going to align with the ones (also comparing with cobra)